packages/react-components/src/components/AttachmentDownloadCards.tsx

```diff
@@ -157,18 +127,22 @@ export const _AttachmentDownloadCards = (props: _AttachmentDownloadCardsProps):
   );
 };
 
-/**
- * @private
- */
-const DownloadIconTrampoline = (): JSX.Element => {
-  // @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload)
-  return <ArrowDownload20Regular data-ui-id={'file-download-card-download-icon'} />;
-  // Return _some_ available icon, as the real icon is beta-only.
-  return <Icon iconName="EditBoxCancel" style={actionIconStyle} />;
-};
-
 const useLocaleStringsTrampoline = (): _AttachmentDownloadCardsStrings => {
   /* @conditional-compile-remove(attachment-download) @conditional-compile-remove(attachment-upload) */
   return useLocale().strings.messageThread;
   return { downloadAttachment: '', attachmentCardGroupMessage: '' };
 };
+
+const getDefaultMenuActions = (chatMessage?: ChatMessage): AttachmentMenuAction[] => {
+  // if message is sent by a Teams user, we need to use a different icon ("open")
+  if (chatMessage?.senderId?.includes('9:orgid')) {
```

Comment:
This seems fragile; should this be extracted to a function in the common packlet? 
Could a teams user also be authenticated and sending a link to some other location?
Should we look at the URL of the linked file?

Reply:
in Teams, sharepoint is the only location where the file can be coming from. that being said, for channel chat, they can have custom storage such as google drive but we don't support channel chat right now.


i feel like url-based on logic is bit of hacky? let me see if there's a more elegant way to check if user is a teams user in ui library